Source/Fuse.Controls.Native/Android/SystemScreenConfigAndroidImpl.uno

```diff
@@ -0,0 +1,104 @@
+using Fuse.Controls.Native;
+
+namespace Fuse.Controls.Native
+{
+	extern(Android) internal class SystemScreenConfigAndroidImpl
+	{
+		private static SystemUiVisibility.Flag _wantedFlags = SystemUiVisibility.Flag.None;
+		private static SystemScreenConfig config;
+
+		public static void Init(SystemScreenConfig config)
+		{
+			SystemScreenConfigAndroidImpl.config = config;
+			SystemUiVisibility.VisibilityChanged += VisibilityChanged;
+		}
+
+		public static void DeInit()
+		{
+			SystemUiVisibility.VisibilityChanged -= VisibilityChanged;
+			SystemScreenConfigAndroidImpl.config = null;
+		}
+
+		public static void TimerDone()
+		{
+			SystemUiVisibility.Flags = _wantedFlags;
+		}
+
+		public static void SetShowState(SystemScreenConfig.Visibility state)
+		{
+			switch(state)
+			{
+				case SystemScreenConfig.Visibility.None:
+					SystemUiVisibility.Flags = SystemUiVisibility.Flag.Fullscreen | SystemUiVisibility.Flag.HideNavigation;
+					_wantedFlags = SystemUiVisibility.Flag.Fullscreen | SystemUiVisibility.Flag.HideNavigation;
+				break;
+				case SystemScreenConfig.Visibility.Minimal:
+					SystemUiVisibility.Flags = SystemUiVisibility.Flag.HideNavigation;
+					_wantedFlags = SystemUiVisibility.Flag.HideNavigation;
+				break;
+				case SystemScreenConfig.Visibility.Full:
+					SystemUiVisibility.Flags = SystemUiVisibility.Flag.None;
+					_wantedFlags = SystemUiVisibility.Flag.None;
+				break;
+			}
+		}
+
+		public static void SetStatusState(bool val)
+		{
+			if(val)
+			{
+				_wantedFlags = _wantedFlags ^ SystemUiVisibility.Flag.Fullscreen;
+				SystemUiVisibility.Flags = SystemUiVisibility.Flags ^ SystemUiVisibility.Flag.Fullscreen;
```

Comment:
nice xor catch, fixed it.

`_wantedFlags` represents the flag state the app wants. The actual value of `Flags` may change due to outside factors, such as the user swiping down from the top while the status bar is hidden.

This is corrected using the timer mechanic, as it is according to @nekronos what most apps do.